evalml/pipelines/components/transformers/preprocessing/transform_primitive_components.py

```diff
@@ -73,17 +73,6 @@ def transform(self, X, y=None):
         es = self._make_entity_set(X_ww)
         features = ft.calculate_feature_matrix(features=self._features, entityset=es)
 
-        # Convert to object dtype so that pd.NA is converted to np.nan
-        # until sklearn imputer can handle pd.NA in release 1.1
-        # FT returns these as string types, currently there isn't much difference
-        # in terms of performance between object and string
-        # see https://pandas.pydata.org/docs/user_guide/text.html#text-data-types
-        # "Currently, the performance of object dtype arrays of strings
-        # "and arrays.StringArray are about the same."
-        features = features.astype(object, copy=False)
-        features.index = X_ww.index
-        features.ww.init(logical_types={col_: "categorical" for col_ in features})
```

Comment:
This popped up as a problem in this CI run - by removing `is_categorical_actually_boolean` usage in the simple imputer, we were no longer recognizing features made from `IsFreeEmailDomain` as boolean columns and imputing them as such in `test_imputer_can_impute_features_generated_from_null_email_url_features`.

I think we're good to remove this block of code for two reasons
1. Freddy added much of that in [this PR](https://github.com/alteryx/evalml/pull/3388/files). We should be good to remove it now that we’re upgraded past sklearn 1.1
2. The fact that we were converting all columns to `categorical` logical type at all was originally done because of a [change in woodwork inference](https://github.com/alteryx/evalml/pull/2690/files#r699592236), but we were also not using woodwork in featuretools at that time (which you can see from the usage of entity in the featuretools calls from that PR), so we no longer have to rely on woodwork inference to get the types of features, which will have woodwork types specified by the primitives used to make them.

Reply:
While sklearn 1.1 did fix the problem Freddy described, I found that removing this block of code entirely caused an issue in Catboost that I was seeing during #3966 with string categories. Converting to `object` fixed that problem, so I'm leaving that as is and will open a ticket to remove this once that bug is fixed.

Reply:
Ticket to remove this workaround: https://github.com/alteryx/evalml/issues/4051